lumigator/python/mzai/backend/backend/services/jobs.py

```diff
@@ -0,0 +1,315 @@
+
+        # set storage path
+        storage_path = f"s3://{Path(settings.S3_BUCKET) / settings.S3_JOB_RESULTS_PREFIX}/"
+
+        # fill up model url with default openai url
+        if request.model.startswith("oai://"):
+            model_url = settings.OAI_API_URL
+        elif request.model.startswith("mistral://"):
+            model_url = settings.MISTRAL_API_URL
+        else:
+            model_url = request.model_url
+
+        # provide a reasonable system prompt for services where none was specified
+        if request.system_prompt is None and not request.model.startswith("hf://"):
+            request.system_prompt = settings.DEFAULT_SUMMARIZER_PROMPT
+
+        config_params = {
+            "job_name": request.name,
+            "job_id": record.id,
+            "model_path": request.model,
+            "dataset_path": dataset_s3_path,
+            "max_samples": request.max_samples,
+            "storage_path": storage_path,
+            "model_url": model_url,
+            "system_prompt": request.system_prompt,
+        }
+
+
+        # load a config template and fill it up with config_params
+        if request.config_infer_template is not None:
+            config_template = request.config_infer_template
+        elif request.model in config_templates.config_infer_template:
+            # if no config template is provided, get the default one for the model
+            config_template = config_templates.config_infer_template[request.model]
+        else:
+            # if no default config template is provided, get the causal template
+            # (which works with seq2seq models too except it does not use pipeline)
+            config_template = config_templates.causal_infer_template
+
+        infer_config_args = {
+            "--config": config_template.format(**config_params),
+        }
+
+        infer_command = f"{settings.LD_PRELOAD_PREFIX} python -m inference infer huggingface"
+
+        # Prepare the job configuration that will be sent to submit the ray job.
+        # This includes both the command that is going to be executed and its
+        # arguments defined in eval_config_args
+        ray_config = JobConfig(
+            job_id=record.id,
+            job_type=JobType.INFERENCE,
+            command=infer_command,
+            args=infer_config_args,
+        )
+
+        # build runtime ENV for workers
+        runtime_env_vars = {"MZAI_JOB_ID": str(record.id)}
+        settings.inherit_ray_env(runtime_env_vars)
+
+        # set num_gpus per worker (zero if we are just hitting a service)
+        if not request.model.startswith("hf://"):
+            worker_gpus = settings.RAY_WORKER_GPUS_FRACTION
+        else:
+            worker_gpus = settings.RAY_WORKER_GPUS
+
+        runtime_env = {
+            "pip": settings.PIP_REQS,
+            "working_dir": settings.EVALUATOR_WORK_DIR,
+            "env_vars": runtime_env_vars,
```

Comment:
A heads up that we'll have different `PIP_REQS`, `EVALUATOR_WORK_DIR`, and (well, of course) `infer_command` here.
How would you prefer to proceed? Happy to send a PR to this branch so we can test everything together, or wait for this to be merged and send a PR to main later on

Reply:
Let's merge this for now and change it in the next one so we can fully test it, does that work for you?

Reply:
Absolutely, go with it and I can change it (well I have done it already :-)) in my PR!